spec/unit/provider/group/groupadd_spec.rb

```diff
@@ -65,12 +67,52 @@
 
   end
 
+  describe "#modify" do
+    before do
+       provider.stubs(:exists?).returns(true)
+    end
+    describe "on systems with the libuser and forcelocal=true" do
```

Comment:
Thank you for adding these new tests! They're a fantastic addition

I'd like to also see tests here making sure we're using `groupmod` when `forcelocal=false`, and that it correctly passes `-o`  when allowdupe is true, also that it errors out if the gid already exists. I think something similar for `#delete` would also be good. The same applies for the useradd spec.

So, my thoughts here are we have a whole matrix of potential outcomes for these cases. I don't want to be too heavy handed and add all potential scenarios, but it would be nice to see a good distribution of test cases for add/modify/delete where forcelocal is true/false and allowdupe is true/false. This would be for both useradd and groupadd.

Reply:
One condition I can't get to work is `has_entry(:custom_environment, has_key('LIBUSER_CONF'))` for the various local commands.  The tests where it was already added work just fine but I'm unable to add that bit of code to test that all `forcelocal` commands are failing.  Example

Works:
```ruby
      it "should use lgroupmod instead of groupmod" do
        provider.expects(:execute).with(['/usr/sbin/lgroupmod', '-g', 150, 'mygroup'])
        provider.gid = 150
      end
```

Fails:
```ruby
      it "should use lgroupmod instead of groupmod" do
        provider.expects(:execute).with(['/usr/sbin/lgroupmod', '-g', 150, 'mygroup'], has_entry(:custom_environment, has_key('LIBUSER_CONF')))
        provider.gid = 150
      end
```